### DISCUSSION TITLE
Expose "operations_research::RoutingSearchParameters" to Python

### DISCUSSION THREAD

**Discussion by hklarnerben** [google/or-tools#3392] [Feature requests] [answered]
**What language and solver does this apply to?**
Python

**Describe the problem you are trying to solve.**
I want to use type hints for the RoutingSearchParameters object.

**Describe the solution you'd like**
Expose "operations_research::RoutingSearchParameters" to pywrapcp:
```
from ortools.constraint_solver.pywrapcp import RoutingSearchParameters
``` 

EDIT:
I want to use type hints:

```python
from ortools.constraint_solver.pywrapcp import RoutingSearchParameters

def my_function(params: RoutingSearchParams):
    ...
```

**Comment by Mizux** [comment]
not sure to exactly understand it, currently you can do:
```py
search_parameters = pywrapcp.DefaultRoutingSearchParameters()
```
but you want IDE completion on `search_parameters.` ?

Concerning Type, our python SWIG  wrapper do not support it (not even sure SWIG support type annotation)

**Comment by Mizux** [comment]
seems related to 
* swig/swig#735

and related to:
* #3182

**Comment by hklarnerben** [comment]
@Mizux I want to use type hints. I edited my issue description above.

**Comment by hklarnerben** [comment]
@Mizux it works for example for `Assignment`:

```python
from ortools.constraint_solver.pywrapcp import Assignment

def myfunc(assignment: Assignment):
    ...
```

I think `RoutingSearchParameters` just needs to be "exposed" to Python like `Assignment` is.

**Comment by Mizux** [comment, marked as answer]
`Assignment` is a C++ class while `RoutingSearchParameters` is a proto message from `routing_parameters.proto`
see: https://github.com/google/or-tools/blob/c78f83e4d4a21059f6e0d6c616e267c5c70d0828/ortools/constraint_solver/routing_parameters.proto#L32-L39

So I would try:
```py
from ortools.constraint_solver.routing_parameters_pb2 import RoutingSearchParameters
def my_function(params: RoutingSearchParameters):
    ...
```

**Reply by hklarnerben** [reply]
Ha, you're right. Works.
